code/pages/ExtensibleSearchPage.php

```diff
@@ -719,11 +748,76 @@ public function getSearchResults($data = null, $form = null) {
 
 		$startTime = microtime(true);
 
+		if($results = $this->getSearchEngineResults($data, $form)) {
```

Comment:
Maybe I should have capitalised "Type" in my comment :D. By 'implemented as a search engine type', I'm meaning to actually pull the existing full-text search logic into a separate class (introducing `interface ExtensibleSearchEngine` probsibly), and change things around from being 'is something applied as an extension with a particular name' to being retrieved as a dependency via DI. Completely removing the ambiguity of where the code should exist relative current extension detection logic. So in the interim yes, the fulltext stuff should be moved into `getSearchEngineResults` in case there's no applied SearchEngine extension.

So going back to

> This would also raise the question, should there be a full-text fallback if the current search engine fails to return any results?

I'd say no, as the 'fallback' type behaviour shouldn't happen automatically.